add more deets to resource version docs (#2728)
pulumi/pulumi-hugo: themes/default/content/docs/intro/concepts/resources/options/version.md

    weight: 15
---

> The `version` resource option specifies a provider version to use when operating on a resource. This version overrides the version information inferred from the current package. This option was built to be used directly by the Pulumi SDK. This option should rarely be used by people directly.

Reviewer: Optional Nit: "rarely" is confusing. We don't mean "don't do it often". We mean "You should know what your doing when you use this, and want to override default resource provider mapping in an untested way."
```suggestion
The `version` resource option specifies a provider version to use when operating on a resource. This version overrides the version information inferred from the current package. This option was built to be used directly by the Pulumi SDK. `version` should not be used directly during normal operations.
```
